***Para acessar os arquivos JSON pelo notebook***

In [1]:
import os

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
!cp -r '/content/drive/MyDrive/DODF_Watcher_JSONs/' 'JSONS'

In [4]:
path_JSONS = 'JSONS'

JSON_dict = {}
JSON_dict['JSON'] = []

list_JSON = os.listdir(path_JSONS)

for JSON in list_JSON:
  JSON_dict['JSON'].append(JSON)

In [8]:
JSON_dict['JSON'][:5]

['DODF 059 28-03-2022.json',
 'DODF 207 05-11-2021.json',
 'DODF 119 28-06-2022.json',
 'DODF 075 25-04-2022.json',
 'DODF 054 21-03-2022.json']

***Tratamento dos JSON***

In [24]:
import json
import re

In [25]:
def pre_process(text):
  text = re.sub(r'<[^\>]*>', '', text)
  return text

*Análise dos Títulos e Tipos de Atos*

In [ ]:
TITULO_dict = {}
TIPO_dict = {}

DOCUMENT_dict = {}
DOCUMENT_dict['TIPO'] = []
DOCUMENT_dict['TITULO'] = []
DOCUMENT_dict['ATO'] = []


for JSON in JSON_dict['JSON']:
  try:

    f = open(path_JSONS + '/' + JSON)
    data = json.load(f)

    for ORGAO in data['json']['INFO']['Seção III']:
      for DOC in data['json']['INFO']['Seção III'][ORGAO]['documentos']:

        DOCUMENT_dict['TIPO'].append(data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['tipo'])
        DOCUMENT_dict['TITULO'].append(data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['titulo'])
        DOCUMENT_dict['ATO'].append(pre_process( data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['texto'] ))


        if data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['titulo'] in TITULO_dict:
          TITULO_dict[data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['titulo']] += 1
        else:
          TITULO_dict[data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['titulo']] = 1
        
        if data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['tipo'] in TIPO_dict:
          TIPO_dict[data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['tipo']] += 1
        else:
          TIPO_dict[data['json']['INFO']['Seção III'][ORGAO]['documentos'][DOC]['tipo']] = 1

    print(f"{JSON} Feito!")

    f.close()

  except:
    print(f"Erro na abertura do JSON: {JSON}")

*Tipos de Atos e a quantidade de suas ocorrências*

Foco: 


*   Extrato
*   Aviso
*   Pregão
*   Extrato de ata
*   Contrato
*   Licitação
*   Aviso de licitação
*   Extrato de contrato





In [13]:
TIPO_dict

{'Extrato': 21487,
 'Edital': 2657,
 'Aviso': 5603,
 'Convocação': 172,
 'Ratificação': 1363,
 'Reconhecimento de dívida': 2105,
 'Ato': 393,
 'Aplicação de penalidade': 474,
 'Julgamento': 42,
 'Relatório': 7,
 'Pregão': 650,
 'Demonstrativo': 59,
 'Ineditorial': 1403,
 'Extrato de ata': 134,
 'Retificacão': 118,
 'Termo aditivo': 333,
 'Habite-se': 64,
 'Concorrência': 63,
 'Declaração': 76,
 'Termo': 541,
 'Contrato': 234,
 'Chamamento': 29,
 'Licitação': 37,
 'Notificação': 294,
 'Chamada pública': 47,
 'Ata': 56,
 'Demonstrativo de despesa': 6,
 'Aviso de licitação': 10,
 'Decisão': 3,
 'Extrato de contrato': 57,
 'Citação por edital': 14,
 'Cancelamento': 11,
 'Errata': 1,
 'Autorização': 11,
 'Plano anual': 13,
 'Despacho': 7,
 'Anulação': 3,
 'Portaria': 3,
 'Dispensa de licitação': 6,
 'Ordem de serviço': 1,
 'Resolução': 3,
 'Tomada de contas': 1,
 'Portaria conjunta': 1,
 'Informação': 1,
 'Apostilamento': 2,
 'Republicação': 1}

In [20]:
import pandas as pd

In [27]:
JSON_df = pd.DataFrame(DOCUMENT_dict)

In [84]:
EXTRATO_df = JSON_df.query("TIPO == 'Extrato'")

In [86]:
EXTRATO_df.head(5)

TIPO                                             TITULO  \
0  Extrato  EXTRATO DO 3º TERMO ADITIVO DO CONTRATO Nº 01/...   
1  Extrato  EXTRATO DO 1º TERMO ADITIVO DO CONTRATO Nº 01/...   
2  Extrato  EXTRATO DO PRIMEIRO TERMO ADITIVO AO CONTRATO ...   
4  Extrato              EXTRATO DO TERMO DE ADESÃO Nº 18/2022   
6  Extrato    EXTRATO DA ATA DE REGISTRO DE PREÇOS Nº 99/2022   

                                                 ATO  
0  Processo: 00136-00000070/2022-64. DAS PARTES: ...  
1  Processo: 00136-00000082/2022-99. DAS PARTES: ...  
2  Processo: 00147-00000138/2020-88. Partes: DF/R...  
4  Processo: 00040-00011296/2022-69 – A SECRETARI...  
6  Processo: 00040-00033745/2021-49, Pregão Eletr...

In [45]:
AVISO_df = JSON_df.query("TIPO == 'Aviso'")

In [81]:
AVISO_df.head(5)

TIPO                                             TITULO  \
5   Aviso                                 AVISO DE LICITAÇÃO   
16  Aviso  AVISO DE SUSPENSÃO DO PREGÃO ELETRÔNICO Nº 19/...   
17  Aviso                                 AVISO DE LICITAÇÃO   
18  Aviso                                 AVISO DE LICITAÇÃO   
19  Aviso                               AVISO AOS ACIONISTAS   

                                                  ATO  
5   PREGÃO ELETRÔNICO Nº 34/2022 - UASG 974002A Pr...  
16  O BRB - BANCO DE BRASÍLIA S.A, torna pública a...  
17  A BRB Serviços S/A torna público o Pregão Elet...  
18  A BRB Serviços S/A torna público o Pregão Elet...  
19  Encontram-se à disposição dos Senhores Acionis...

In [50]:
AVISO_DE_LICITACAO_df = JSON_df.query("TIPO == 'Aviso de licitação'")

In [77]:
AVISO_DE_LICITACAO_df.head(5)

TIPO                          TITULO  \
1221   Aviso de licitação  AVISO DE ABERTURA DE LICITAÇÃO   
1222   Aviso de licitação  AVISO DE ABERTURA DE LICITAÇÃO   
1223   Aviso de licitação  AVISO DE ABERTURA DE LICITAÇÃO   
2353   Aviso de licitação   AVISO DE LICITAÇÃO FRACASSADA   
11867  Aviso de licitação              AVISO DE LICITAÇÃO   

                                                     ATO  
1221   O Presidente da Comissão de Licitação para Com...  
1222   O Presidente da Comissão de Licitação para Com...  
1223   O Presidente da Comissão de Licitação para Com...  
2353   PREGÃO ELETRÔNICO Nº 03/2022A SECRETARIA DE ES...  
11867  PREGÃO ELETRÔNICO Nº 06/2022Objeto: Fornecimen...

In [55]:
PREGAO_df = JSON_df.query("TIPO == 'Pregão'")

In [75]:
PREGAO_df.head(5)

TIPO                                    TITULO  \
124  Pregão              PREGÃO ELETRÔNICO Nº 09/2022   
182  Pregão  RESULTADO - PREGÃO ELETRÔNICO Nº 58/2021   
183  Pregão  RESULTADO - PREGÃO ELETRÔNICO Nº 64/2021   
204  Pregão                   RESULTADO DE JULGAMENTO   
353  Pregão                   RESULTADO DE JULGAMENTO   

                                                   ATO  
124  A SECRETARIA DE ESTADO DE CULTURA DO DISTRITO ...  
182  O BRB - BANCO DE BRASÍLIA S.A. torna público o...  
183  Objeto: serviços de manutenção preventiva e co...  
204  PREGÃO ELETRÔNICO POR SRP Nº 373/2021 - UASG 9...  
353  PREGÃO ELETRÔNICO Nº 128/2022 - UASG 926119O P...

In [61]:
CONTRATO_df = JSON_df.query("TIPO == 'Contrato'")

In [72]:
CONTRATO_df.head(5)

TIPO                                             TITULO  \
395  Contrato  CONTRATO DE PRESTAÇÃO DE SERVIÇOS Nº 25/2022, ...   
494  Contrato                                CONTRATO Nº 09/2022   
727  Contrato      CONTRATO DE PRESTAÇÃO DE SERVIÇOS Nº 09/2022,   
961  Contrato                                ADITIVO DE CONTRATO   
962  Contrato                                ADITIVO DE CONTRATO   

                                                   ATO  
395  Processo: 00150-00002033/2022-85.O DISTRITO FE...  
494  Contrato nº 09/2022 - TCB/FINO SABOR INDÚSTRIA...  
727  NOS TERMOS DO PADRÃO Nº 01/2002Processo: 00150...  
961  Processo: 00311-00000006/2020-34. A Diretoria ...  
962  Processo: 00311-00000007/2021-60. A Diretoria ...

In [63]:
LICITACAO_df = JSON_df.query("TIPO == 'Licitação'")

In [70]:
LICITACAO_df.head(5)

TIPO                                             TITULO  \
493   Licitação                      RESULTADO FINAL DO JULGAMENTO   
686   Licitação                             RESULTADO DE LICITAÇÃO   
1097  Licitação                             RESULTADO DE LICITAÇÃO   
1740  Licitação  LICITAÇÃO PÚBLICA PARA CONCESSÃO DE DIREITO RE...   
3606  Licitação                             RESULTADO DE LICITAÇÃO   

                                                    ATO  
493   TOMADA DE PREÇOS Nº 01/2021Processo: 113-00019...  
686   PREGÃO ELETRÔNICO INTERNACIONAL Nº 03/2021Proc...  
1097  A Caesb torna público o resultado do julgament...  
1740  AVISO DE EDITAL Nº 10/2021A COMPANHIA IMOBILIÁ...  
3606  A Caesb torna público o resultado do julgament...

***Análise de Tipos:***


1.   Licitação

*   ***Pode indicar a Reabertura de uma licitação passada, porém na maioria indica resultados de licitações***

2.   Contrato

*   ***Licitação já assinada ou algum aditivo de contrato***

3.   Pregão

*   ***Pode contar a indicação de abertura de uma nova licitação, mas na maior parte contém os resultados de licitações***

4.   Aviso de Licitação

*   ***Pode indicar um aviso de licitação fracassada, mas na maioria dos casos indica uma abertura ou futura abertura de licitação***

5.   Aviso

*   ***Na maioria dos casos indica uma abertura ou reabertura de licitação, mas pode ser também de resultado ou suspensão***

6.   Extrato

*   ***Pode indicar aditivos, dentre outros mas tem presente a maior parte das entidades presentes em uma licitação***


***Tipos de Entidades:***


1.   Extrato de Licitação

*   Processo: ***Processo: 00136-00000070/2022-64***

*   Partes do Ato: ***DAS PARTES: Administração Regional do Núcleo Bandeirante - RA-NB e Fundação de Amparo ao Trabalhador Preso do Distrito Federal - FUNAP/DF***

*   CNPJ de alguma das Partes: ***CNPJ nº 04.744.995/0001-56***

*   Objeto do Ato: ***DO OBJETO: Prorrogação da vigência do contrato por mais 12 (doze) meses***

*   Lei: ***inciso II do art. 57 da Lei nº 8.666/93***

*   Valor: ***VALOR: R$ 450.000,00 (quatrocentos e cinquenta mil reais)***

*   Dotação Orçamentária: ***DOTAÇÃO ORÇAMENTÁRIA: Unidade Orçamentária: 09110***

*   Programa de Trabalho: ***Programa de Trabalho: 04.421.6217.2426.0052 e 04.421.6217.2426.8567***

*   Natureza da Despesa: ***Natureza da Despesa: 33.91.39***

*   Fonte de Recursos: ***Fonte de Recursos: 100***

*   Nota de Empenho: ***Nota de Empenho: 2022NE00040***

*   Modalidade: ***Modalidade: Estimativo***

*   Data da Assinatura: ***DATA DA ASSINATURA: 09 de março de 2022***

*   Vigência: ***VIGÊNCIA: 21/03/2022 até 20/03/2023***

*   Signatários: ***SIGNATÁRIOS: Pelo Distrito Federal, ADALBERTO FERREIRA DE PAULA CARVALHO, na qualidade de Administrador Regional do Núcleo Bandeirante; Pela Contratada, DEUSELITA PEREIRA MARTINS, na qualidade de representante legal***




2.   Aviso de Licitação

*   Número do Pregão: ***PREGÃO ELETRÔNICO Nº 34/2022 - UASG 974002***

*   Licitação: ***licitação do PE 34/2022***

*   Objeto do Ato: ***cujo objeto é o Registro de Preços para eventual contratação de solução de TIC parametrizável para fiscalização eletrônica de mercadorias em veículos em trânsito nas estradas e rodovias do Distrito Federal, compreendendo serviços de coleta e tratamento de dados, implantação, operação e suporte técnico da solução, a fim de atender às necessidades da Secretaria de Estado de Economia do Distrito Federal, de acordo com as condições e características constante no Edital e seus anexos***

*   Valor: ***Valor estimado: R$ 53.739.224,10***

*   Tipo de Licitação: ***Tipo de Licitação: Menor Preço.***

*   Elemento de Despesa: ***Elemento de Despesa: 44.90.52***

*   Data de Abertura: ***Abertura das propostas dia 08/04/2022, às 9h***

*   Processo: ***Processo 00040-00029314/2019-63***

*   Tipo de Licitação: ***Tipo de Licitação: Menor Preço.***

*   Pregoeira/Pregoeiro: ***THAIS RORIZ AZEVEDOPregoeira***





